src/ffi.rs

```diff
@@ -543,6 +546,40 @@ pub extern fn quiche_conn_send(
     }
 }
 
+#[cfg(unix)]
+#[repr(C)]
+pub struct send_info {
+    pub send_time: timespec,
+}
+
+#[no_mangle]
+#[cfg(unix)]
+pub extern fn quiche_conn_send_with_info(
+    conn: &mut Connection, out: *mut u8, out_len: size_t,
+    out_info: &mut send_info,
+) -> ssize_t {
+    if out_len > <ssize_t>::max_value() as usize {
+        panic!("The provided buffer is too large");
+    }
+
+    let out = unsafe { slice::from_raw_parts_mut(out, out_len) };
+
+    match conn.send_with_info(out) {
+        Ok((v, info)) => {
+            unsafe {
+                ptr::copy_nonoverlapping(
+                    &info.send_time as *const _ as *const timespec,
```

Comment:
Related to the comment above on `#[cfg(unix)]` -- this should be done only for the platform which timespec is used.
Also this is more like a question, how do we make sure we can safely convert `Instant` to `timespec`? Is `cfg()` check is enough?

Reply:
yes, it should be done where timespec is valid. Changed to include in Linux platforms only.
There is no safe/proper way. With 'linux only' check this should be ok I believe.